Docs: new section for contributing (#11812)
gatsbyjs/gatsby: docs/contributing/docs-contributions.md

---
title: Docs Contributions
---

> Gatsby, unsurprisingly, uses Gatsby for its documentation website. Thank you in advance and cheers for contributing to Gatsby documentation! It's people like you that make this community great!

Reviewer: It might be useful to highlight the number of people who have already done this, to reassure folks that this is a pretty common thing to do.

`git log --pretty="format:%ae" docs/` yields 870 unique emails, so we could say, eg "As of February 2019, over 800 people have contributed to the Gatsby documentation" or whatever makes sense.

3. GitHub then allows you to commit the change and raise a PR right in the UI. This is the _easiest_ way you can contribute to the project!

#### Converting a document from a stub

If you wrote a new document that was [previously a stub](/contributing/how-to-write-a-stub/), there are two things you need to update.

1. Remove the frontmatter that links to the issue

```diff:title=docs/docs/example-doc.md
  ...
    title: Example Document
- - issue: https://github.com/gatsbyjs/gatsby/issues/00000
+ -
  ...
```

2. Edit `www/src/data/sidebars/doc-links.yaml` by removing the asterisk behind the document's title:

```diff:title=www/src/data/sidebars/doc-links.yaml
  ...
- - title: Example Document*
+ - title: Example Document
    link: /docs/example-document/
  ...
```

Want more? Check out our additional pages on docs contributions:

- [Gatsby Style Guide](/contributing/gatsby-style-guide/)
- [Docs Templates](/contributing/docs-templates/)

> - [How to Write a Stub](/contributing/how-to-write-a-stub/)

Reviewer: Might be worth saying something along the lines of "after your first code contribution to the Gatsby repo (including documentation) you become eligible for a free shirt or pair of socks. See the swag page for more details!"

Author: I added a section for it above these links!

---
title: Docs Contributions
---

> Gatsby, unsurprisingly, uses Gatsby for its documentation website. Thank you in advance and cheers for contributing to Gatsby documentation! As of February 2019, over 800 people have contributed. It's people like you that make this community great!

Reviewer 2: Was this number targeted somehow to just docs contributions? Otherwise, it's about double!

```suggestion
Gatsby, unsurprisingly, uses Gatsby for its documentation website. Thank you in advance and cheers for contributing to Gatsby documentation! As of February 2019, over 1600 people have contributed. It's people like you that make this community great!
```

## Options for contributing to the Gatsby docs

When working on the Gatsby.js documentation, you can choose between two major styles of working:

- [Work directly in the GitHub UI](#modifying-markdown-files), using the "Edit this File" and commit capabilities without having to clone the repository. This is useful for quick documentation updates, typo fixes, and lightweight Markdown changes.
- Clone the Gatsby.js repo and get the `www` site up and running locally. This is necessary for more thorough documentation content and infrastructure changes. Learn how to get set up in the [Gatsby docs setup instructions](/contributing/setting-up-your-local-dev-environment#docs-site-setup-instructions).

> ### Modifying markdown files

Reviewer 2: Nit: but I think it's the preferred capitalization

```suggestion
### Modifying Markdown files
```

1. If you want to add/modify any Gatsby documentation, go to the
   [docs folder](https://github.com/gatsbyjs/gatsby/tree/master/docs) or [contributing folder](https://github.com/gatsbyjs/gatsby/tree/master/contributing) on GitHub and
   use the file editor to edit and then preview your changes.
2. Before committing the change and raising a PR in the UI, you need to make sure the PR meets the docs contribution criteria:

>    - Follow the standards outlined in the [Gatsby Style Guide](https://www.gatsbyjs.org/contributing/gatsby-style-guide/).

Reviewer 2: ```suggestion
   - Follow the standards outlined in the [Gatsby Style Guide](/contributing/gatsby-style-guide/).
```

>    - If your PR did not come from an issue written by the core team, please add a comment to your PR that explains why it should be included in the docs, according to the [Docs Decision Tree](https://www.gatsbyjs.org/blog/2018-10-12-uptick-docs-contributions-hacktoberfest/#docs-decision-tree-and-examples).

Reviewer 2: ```suggestion
   - If your PR did not come from an issue written by the core team, please add a comment to your PR that explains why it should be included in the docs, according to the [Docs Decision Tree](/blog/2018-10-12-uptick-docs-contributions-hacktoberfest/#docs-decision-tree-and-examples).
```

>      > Note: If your issue and/or PR doesn't meet the above contribution criteria, it may receive a comment reminding you to do so. If, after two weeks, these updates haven't been made, your issue and/or PR may be closed, which helps us triage issues and PRs efficiently. You can request that it be reopened if and when you are ready to make the updates required.

Reviewer 2: We don't _currently_ always follow this, but if this is something we should be doing--happy to accommodate our workflow!

Author: I think this was ported from the old doc, should we update it to say something more accurate for the current workflow?

Reviewer 2: Maybe it's something the docs team previously followed and core didn't?

But yeah - this should stay if that's the case! The nice thing is that if we're explicit in the two weeks, we can use automated tooling to just do this for us (e.g. once we add a certain label, two weeks from then our bot will close).

Author: @jlengstorf or @shannonbux do you have any insight into the above note? I'm not sure if it's accurate. Hopefully it won't hold up this PR since that content was ported from the old docs page.

3. GitHub then allows you to commit the change and raise a PR right in the UI. This is the _easiest_ way you can contribute to the project!

#### Converting a document from a stub

If you wrote a new document that was [previously a stub](/contributing/how-to-write-a-stub/), there are two things you need to update.

1. Remove the frontmatter that links to the issue

```diff:title=docs/docs/example-doc.md
  ...
    title: Example Document
- - issue: https://github.com/gatsbyjs/gatsby/issues/00000
+ -
  ...
```

2. Edit `www/src/data/sidebars/doc-links.yaml` by removing the asterisk behind the document's title:

```diff:title=www/src/data/sidebars/doc-links.yaml
  ...
- - title: Example Document*
+ - title: Example Document
    link: /docs/example-document/
  ...
```

### Claim your swag

> After your first code contribution to the Gatsby repo (including documentation) you become eligible for a free shirt or pair of socks. See the [swag page](/contributing/contributor-swag/) for more details!

Reviewer 2: ```suggestion
After your first contribution to the Gatsby repo is merged you become eligible for a free shirt or pair of socks. See the [swag page](/contributing/contributor-swag/) for more details!
```

Reviewer 2: It might be worth keeping the (including documentation) parenthetical 🤔

Reviewer 2: Ignore - missed the comment from @calcsam!
